misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/JDIEventProcessor.java

```diff
@@ -121,6 +129,20 @@ private void processEvent(EventSet eventSet, Event event) {
                 || event instanceof VMDeathEvent
                 || event instanceof VMDisconnectedException) {
             isRemoteVmAttached = false;
+        } else if (event instanceof ThreadStartEvent threadStartEvent) {
+            ThreadReference thread = threadStartEvent.thread();
+            if (thread.isVirtual()) {
+                synchronized (virtualThreads) {
+                    virtualThreads.add(thread);
+                }
+            }
+            eventSet.resume();
+        } else if (event instanceof ThreadDeathEvent threadDeathEvent) {
+            ThreadReference thread = threadDeathEvent.thread();
+            synchronized (virtualThreads) {
+                virtualThreads.remove(thread);
+            }
+            eventSet.resume();
```

Comment:
For the `virtualThreads` variable, using a `ConcurrentLinkedQueue` or `CopyOnWriteArrayList` (concurrent data structures) is generally better than using an ArrayList with explicit synchronization blocks.